python/lsst/ts/hvac/xml/hvac_mqtt_to_SAL_XML.py

```diff
@@ -19,153 +19,495 @@
+                        unit = "%"
+                    else:
+                        raise ValueError(f"Unknown unit {row['unit']}")
+                    topics[hvac_topic.name][hvac_item.name] = {
+                        "idl_type": idl_type,
+                        "unit": unit,
+                    }
+                    hvac_topics[topic_and_item] = {
+                        "idl_type": idl_type,
+                        "unit": unit.strip(),
+                        "topic_type": row["rw"].strip(),
+                        "limits": _parse_limits(row["limits"].strip()),
+                    }
+
                     item_found = True
                     break
             if not item_found:
-                raise ValueError(f"CommandItem '{command}' for {topic} not found.")
+                raise ValueError(f"TelemetryItem '{item}' for {topic} not found.")
 
             topic_found = True
             break
     if not topic_found:
-        raise ValueError(f"Topic {topic} not found.")
+        raise ValueError(f"Topic {topic} in topic_and_item {topic_and_item} not found.")
 
 
 def collect_topics_and_items():
+    """Loop over all rows in the CSV file and extracts either telemetry topic
+    data or command topic data depending on the contents of the 8th column in
+    the CSV row.
+    """
     with open(dat_control_filename) as csv_file:
-        csv_reader = csv.reader(csv_file, delimiter=";")
-        for row in csv_reader:
-            if row[7] == "PUBLICACION":
-                collect_telemetry_topics_and_items(row)
-            if row[7] == "SUSCRIPCION":
-                collect_command_topics_and_items(row)
+        csv_reader = pandas.read_csv(
+            csv_file,
+            delimiter=";",
+            index_col=False,
+            dtype=str,
+            names=names,
+            keep_default_na=False,
+        )
+        for index, row in csv_reader.iterrows():
+            if row["rw"].strip() == "READ":
+                _generic_collect_topics_and_items(row, telemetry_topics, TelemetryItem)
+            if row["rw"].strip() == "WRITE":
+                _generic_collect_topics_and_items(row, command_topics, CommandItem)
 
 
-def translate_item(item):
-    item = re.sub(r"([A-Z])", lambda m: " " + m.group(1), item).upper()
+def _translate_item(item):
+    """Perform a crude translation of the Spanish words in the given item to
+    English.
+
+    Parameters
+    ----------
+    item: `str`
+        A string containing Spanish words to be translated into English.
+
+    Returns
+    -------
+    translated_item: `str`
+        A string containing a crude English translation of the Spanish words.
+
+    """
+    translated_item = re.sub(r"([A-Z])", lambda m: " " + m.group(1), item).upper()
     for key in dictionary.keys():
-        item = re.sub(rf"{key}", rf"{dictionary[key]}", item)
-    return item
+        translated_item = re.sub(rf"{key}", rf"{dictionary[key]}", translated_item)
```

Comment:
Where does `dictionary` come from? It appears to be a global variable, in which case please use uppercase and a more descriptive name. such as SPANISH_ENGLISH_DICT

Reply:
Good suggestion of course and sloppy coding from my side. Done.